src/components/clusterprovider/minikube/minikube.ts

```diff
@@ -75,6 +78,34 @@ class MinikubeImpl implements Minikube {
     status(): Promise<MinikubeInfo> {
         return minikubeStatus(this.context);
     }
+
+    checkUpgradeAvailable() {
+        return minikubeUpgradeAvailable(this.context);
+    }
+}
+
+async function minikubeUpgradeAvailable(context: Context) {
+    if (!await checkPresent(context, CheckPresentMode.Alert)) {
+        // not installed, no upgrade.
+        return;
+    }
+
+    const sr = await context.shell.exec(`${context.binPath} update-check`);
+    const lines = sr.stdout.split(os.EOL);
+    const currentVersion = extractVersion(lines[0]);
+    const availableVersion = extractVersion(lines[1]);
+    if (currentVersion !== availableVersion) {
+        vscode.window.showInformationMessage('Minikube upgrade available.', 'Install').then((value: string) => {
+            if (value === 'Install') {
+                vscode.commands.executeCommand('vscode.open', vscode.Uri.parse('https://github.com/kubernetes/minikube/releases/latest'));
```

Comment:
When we offer an install (via the `Install Dependencies` button) we usually download the dependency, drop it into the extension tools directory, and add a config entry so that the extension uses it.  See `src/components/installer/installer.ts` which already has an `installMinikube` function.  It would be great if we could adapt that function to fetch the latest version rather than whatever version was current when we last updated that code.  _cough cough_

Alternatively, rename this button to something like `View Latest Release`, so it's clear that it opens the release page rather than actually modifying the user's system.

Reply:
Thinking about it, since you know `availableVersion` from the version check, it should be possible to modify the existing `installMinikube` function to take it as an argument and use it when constructing the download URL.  So making this follow the existing 'install dependencies' paradigm might not be too hard after all.

Reply:
done.